### DISCUSSION TITLE
OpenTelemetry - XRay does not clearly represents all the services

### DISCUSSION THREAD

**Discussion by robvanpamel** [aws/aws-dotnet-messaging#143] [Q&A] [answered]
Hi, 

I start to use AWS Messaging in combination with OpenTelemetry and X-Ray. I must admit that i am new to X-Ray so my assumption can be wrong (-used to work with DataDog in the past-).

Let say that i created a producer and a consumer that use SQS as their messaging protocol. Now when the producer publishes a messages, it goes to SQS and their the consumer picks it up to process it. 
What I expected to see in X-Ray is 

Publisher --> Publish Message --> SQS --> Consumer --> Process Message 
However, the semantics of Publisher and Consumer seem to be wiped out although in OpenTelemetry the data is clearly sent to X-Ray (cref: service.name) . 
The problem with this solution is that it gives the impression that the Publisher is doing all the work, and the consumer is not involved.  Let say that the consumer has dependencies to other services, this will cause a lot of confusion.

How can I create a  "X-Ray node" that represent my Consumer / Producer ? 

![image](https://github.com/awslabs/aws-dotnet-messaging/assets/1253868/27e810af-e3c3-4d89-9ff8-f64c3afd1231)

I attached both the logs as well. 

[trace.subscriber.txt](https://github.com/awslabs/aws-dotnet-messaging/files/15416964/trace.subscriber.txt)
[trace.publisher.txt](https://github.com/awslabs/aws-dotnet-messaging/files/15416957/trace.publisher.txt)
[x-ray-trace.txt](https://github.com/awslabs/aws-dotnet-messaging/files/15416965/x-ray-trace.txt)

**Comment by bhoradc** [comment, marked as answer]
Hello @robvanpamel, 

We have converted this discussion into a new issue [#159](https://github.com/awslabs/aws-dotnet-messaging/issues/159) which would be investigated by team.

Regards, 
Chaitanya
